### PR TITLE
Update references to main branch

### PR DESCRIPTION
**Pull request by fran893**
Fix 3663

### Purpose
*What is the context of this pull request? Why is it being done?*
As the **master** will be renamed to **main**, we updated all the references to this branch.

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

We modified the following files:
- Java files
- Markdown files
- YAML files
- JSON files
- XML files
- JavaScript files
- CSS file
- PowerShell file
- Gitignore file
- HTML+Razor file

Furthermore, we updated the references to external repositories and fixed broken links.

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
*-*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*
*-*

### Checklist
*-*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [x] I have updated related documentation

### CODE REVIEW

**Review by Batta32 — CHANGES_REQUESTED** [review]
@fran893 @FedericoBernal FYI, review the requested changes, but I'll apply the fix, this is just FYI

**Review by Batta32 — CHANGES_REQUESTED** [review]

**Review by Batta32 — APPROVED** [review]
LGTM

File: docs/_docs/overview/virtual-assistant-template.md [hunk 1]
```diff
@@ -142,12 +142,7 @@ A simple introduction card is provided as standard which you can adapt as needed
 ### Onboarding Dialog
 {:.no_toc}
 
-The [**OnboardingDialog**]({{site.repo}}/blob/main/templates/csharp/VA/VA/Dialogs/OnboardingDialog.cs) provides an example introduction Dialog experience for users starting their first conversation. It prompts for some information which is then stored in State for future use by your assistant. This dialog demonstrates how you can use prompts and state.
-
-### Escalate Dialog
```
Comment (FedericoBernal) [inline docs/_docs/overview/virtual-assistant-template.md:147]: Why have we deleted this?

File: docs/_docs/overview/virtual-assistant-template.md [hunk 2]
```diff
@@ -254,16 +249,7 @@ The **MainDialog** class provided in the template derives from a base class call
 
 This RouterDialog as part of the **OnContinueDialogAsync** handler invokes on the **OnInterruptDialogAsync** within your **MainDialog.cs**. This handler enables interruption logic to be processed before any utterance is processed, by default Cancel, Help, Logout and Restart are handled as part of this handler enabling top-level intent processing even when you have an active dialog.
 
-You can review this logic within [**MainDialog.cs**]({{site.repo}}/blob/main/templates/csharp/VA/VA/Dialogs/MainDialog.cs#L295).
-
-### Event processing
```
Comment (FedericoBernal) [inline docs/_docs/overview/virtual-assistant-template.md:259]: Same as before

File: docs/_docs/overview/virtual-assistant-template.md (hunk 1 above)
Comment (Batta32) [reply docs/_docs/overview/virtual-assistant-template.md:147]: The `EscalateDialog` doesn't exist in the Virtual Assistant, that's why I deleted it

File: docs/_docs/overview/virtual-assistant-template.md (hunk 2 above)
Comment (Batta32) [reply docs/_docs/overview/virtual-assistant-template.md:259]: Same, the handle of `OnEventAsync` method was deleted in the VA

**Review by FedericoBernal — APPROVED** [review]
LGTM!

**Pull request closed without merging**